src/main/java/sg/ncl/MainController.java

```diff
@@ -2389,50 +2391,56 @@ public String saveExperimentImage(
             @PathVariable String expId,
             @PathVariable String nodeId) throws IOException {
 
-        if (saveImageForm.getImageName().length() < 2) {
+        String REDIRECT_SAVING_IMAGE = "redirect:/experiments/save_image/";
+
+        //checking image name
+        String imageName = saveImageForm.getImageName();
+        if (imageName.length() < 2) {
             log.warn("Save image form has errors {}", saveImageForm);
-            redirectAttributes.addFlashAttribute("message", "Image name too short, minimum 2 characters");
-            return "redirect:/experiments/save_image/" + teamId + "/" + expId + "/" + nodeId;
+            redirectAttributes.addFlashAttribute("message", "Image name is required minimum of 2 characters");
+            return REDIRECT_SAVING_IMAGE + teamId + "/" + expId + "/" + nodeId;
+        } else if (!VALID_IMAGE_NAME.matcher(imageName).matches()) {
+            redirectAttributes.addFlashAttribute("message", "Image name can only contain alphanumeric characters and \"-\"");
+            return REDIRECT_SAVING_IMAGE + teamId + "/" + expId + "/" + nodeId;
         }
 
         log.info("Saving image: team {}, experiment {}, node {}", teamId, expId, nodeId);
-
         ObjectMapper mapper = new ObjectMapper();
         HttpEntity<String> request = createHttpEntityWithBody(mapper.writeValueAsString(saveImageForm));
-
+        ResponseEntity response;
         restTemplate.setErrorHandler(new MyResponseErrorHandler());
-        ResponseEntity response = restTemplate.exchange(properties.saveImage(), HttpMethod.POST, request, String.class);
+        response = restTemplate.exchange(properties.saveImage(), HttpMethod.POST, request, String.class);
         String responseBody = response.getBody().toString();
 
         if (RestUtil.isError(response.getStatusCode())) {
             MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
             ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
 
             log.warn("Save image: error with exception {}", exceptionState);
-
+            String IMAGE_SAVING_ERROR = "Saving image error: {}";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)